### PR TITLE
Reruns Dapr builds a few times, if needed.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Reruns the build workflow a few times if needed, not against pull requests.

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .github/workflows/dapr.yml [hunk 1]
```diff
@@ -686,3 +686,13 @@ jobs:
...
+  rerun:
+    name: Reruns in case of failure
+    needs: helmpublish
+    if: failure() && fromJSON(github.run_attempt) < 3 && github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - env:
+          GH_REPO: ${{ github.repository }}
+          GH_TOKEN: ${{ github.token }}
+        run: gh workflow run rerun.yml -F run_id=${{ github.run_id }}
```
Comment (ItalyPaleAle) [inline .github/workflows/dapr.yml:698]: Rather than doing this, which IMHO adds complexity and it's hard to follow, let's just add autmatic retries to tests when not building a PR (just one CLI flag passed to gotestsum). Although I think this should only be done for releases.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

**Comment by artursouza** [comment]
Keeping it as a draft to collect feedback on this change.

**Comment by artursouza** [comment]
Closing the PR. We can always get back to this if we decide it is something we want to proceed with.
